launcher/cli/logging.go

```diff
@@ -100,6 +104,58 @@ func setupLogger() {
 
 	// Hijack standard Golang `log` and redirect it to our common logger
 	zap.RedirectStdLogAt(commonLogger, zap.DebugLevel)
+
+	if listenAddr != "" {
+		go func() {
+			userLog.Debug("starting atomic level switcher on" + listenAddr)
+			if err := http.ListenAndServe(listenAddr, http.HandlerFunc(handleHTTPLogChange)); err != nil {
+				userLog.Warn("failed listening on :1065 to switch log level:", zap.Error(err))
+			}
+		}()
+	}
+
+}
+
+type logChangeReq struct {
+	Inputs string `json:"inputs"`
+	Level  string `json:"level"`
+}
+
+func handleHTTPLogChange(w http.ResponseWriter, r *http.Request) {
+
+	b, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		http.Error(w, fmt.Sprintf("cannot read body: %s", err), 400)
+		return
+	}
+
+	// Unmarshal
+	var in logChangeReq
+	err = json.Unmarshal(b, &in)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("cannot unmarshal JSON body: %s", err), 400)
+		return
+	}
+
+	if in.Inputs == "" {
+		http.Error(w, fmt.Sprintf("inputs not defined, should be comma-separated list of words or a regular expressions: %s", err), 400)
+		return
+	}
+
+	switch strings.ToLower(in.Level) {
+	case "":
```

Comment:
Why not having a `default` case instead that catch all other invalid inputs at the same time?